tests/vm_test.go

```diff
@@ -2460,45 +2459,105 @@ status:
 				vm, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, controller.GeneratePatchBytes(ops), &metav1.PatchOptions{})
 				Expect(err).ToNot(HaveOccurred())
 			}
-
-			err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
-
 			By("Ensure the vm has disappeared")
 			Eventually(func() bool {
 				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
 				return errors.IsNotFound(err)
 			}, 2*time.Minute, 1*time.Second).Should(BeTrue(), fmt.Sprintf("vm %s is not deleted", vm.Name))
 		})
 
-		It("should be added when the vm is created", func() {
+		It("should be added when the vm is created and removed when the vm is being deleted", func() {
+			vm.Finalizers = append(vm.Finalizers, customFinalizer)
```

Comment:
Since both tests has the `customFinalizer` this can be done in `BeforeEach` section. 
With this, we can also remove the `if controller.HasFinalizer(vm, customFinalizer)` check in the AfterEach.
WDYT?